src/store/modules/stock/actions.ts

```diff
@@ -63,6 +66,167 @@ const actions: ActionTree<StockState, RootState> = {
   clearStockItems({ commit }){
     commit(types.STOCK_ITEMS_UPDATED, { parsed: [], original: []});
   },
+  async processUpdateRestockItems({ commit }, items) {
+  
+    const productIds = items.map((item: any) => item.identification);
+  
+    const payload = {
+      productIds,
+      identificationTypeId: items[0]?.identificationTypeId
+    };
+  
+    const cachedProducts = await store.dispatch("product/fetchProducts", payload);
+    // creating products object based on identification selected
+    const products: any = Object.values(cachedProducts).reduce((updatedProducts: any, product: any) => {
+      const identification = product.identifications.find((identification: any) => payload.identificationTypeId.toLowerCase() === identification.productIdTypeEnumId.toLowerCase())
+      updatedProducts[identification.idValue] = product
+      return updatedProducts;
+    }, {})
+
+    const initial = items.map((item: any) => {
+      const product = products[item.identification];
+      
+      if (product) {
+        item.parentProductId = product?.parent?.id;
+        item.pseudoId = product.pseudoId;
+        item.parentProductName = product?.parent?.productName;
+        item.productId = product.productId
+        item.imageUrl = product.images?.mainImageUrl;
+        item.isSelected = true;
+        return item;
+      }
+      return;
+    }).filter((item: any) => item);
+  
+    commit(types.STOCK_SCHEDULE_ITEMS_UPDATED, initial);
+  },
+  async scheduledStock({ commit }, payload) {
+    commit(types.STOCK_SCHEDULED_INFORMATION, payload)
+  },
+  async clearScheduledStock({ commit }) {
+    commit(types.STOCK_SCHEDULED_INFORMATION, {})
+  },
+  async shopifyShop({ commit }, payload) {
+    commit(types.STOCK_SHOPIFY_SHOPS_UPDATED, payload)
+  },
+  
+  async scheduleService({ dispatch, state }, { params, restockName }) {
+    let resp;
+
+      const job = await dispatch("fetchDraftJob")
+
+      if(!job.jobId || !job.serviceName || job.serviceName == '_NA_') {
+        showToast(translate("Configuration missing"))
+        return;
+      }
+
+      const payload = {
+        'JOB_NAME': restockName || state.schedule.restockName || `Created ${DateTime.fromMillis(+state.schedule.scheduledTime).toLocaleString(DateTime.DATETIME_MED)}`,
```

Comment:
Change the time here to the current time instead of scheduledTime if the restockName is not found.